ext/MitsubaLoader/CMitsubaLoader.cpp

```diff
@@ -1409,20 +1431,50 @@ void instr_execute(in instr_t instr, in uvec3 regs, in mat2 dUV, in vec3 L)
 	uint ndf = instr_getNDF(instr);
 	bool twosided = instr_getTwosided(instr);
 	
+	vec3 scale = decodeRGB19E7(bsdf_data.data[0].xy);
+	vec3 firstParameter;
+	vec3 opacity;
+	if (opcode_isBxDF(opcode))
+	{
+		bool firstParamTexPresence = instr_get1stParamTexPresence(instr);
+		bool opacityTexPresence = instr_getOpacityTexPresence(instr);
+		opacity = textureOrRGB19E7(bsdf_data.data[0].zw, opacityTexPresence, dUV)*scale.z;
+		firstParameter = textureOrRGB19E7(bsdf_data.data[1].xy, firstParamTexPresence, dUV)*scale.x;
```

Comment:
do some reasonable checks if you need to fetch the texture (its okay to diverge fora texture fetch because it will skip the memory fetches, saving bandwidth, its not okay to diverge the [multiply] artihmetic )

Reply:
let me know if tackled in new implementation.

Reply:
?

Reply:
irrelevant, all parameters are fetched without any divergence other than texture fetch or constant choice